pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1767,4 +1769,96 @@ public void testNoParametersNPE() throws SQLException {
       // ignore
     }
   }
+
+  @Test
+  public void testBatchSelect() throws SQLException {
+    PreparedStatement pstmt = null;
+    PreparedStatement batchSelect = null;
+    ResultSet rs = null;
+    try {
+      pstmt = con.prepareStatement("INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);");
+      ((PgStatement) pstmt).setPrepareThreshold(0);
+      for (int i = 0; i < 4; i++) {
```

Comment:
Just wondering: do your production usages include executing statements of various shapes? For instance something like insert-select-insert, "select from a; select from b; select from c"?